### PR TITLE
Fix bug with preview modal header

### PR DESCRIPTION
**Pull request by helenellyx**
## Notion ticket link

[Preview displays “draft” for published reviews](https://www.notion.so/uwblueprintexecs/Preview-displays-draft-for-published-reviews-d736f8cd836b44af8ef54923f8d96e93)

## Implementation description
*  Checked if the published_at data field for Review is undefined or null
* If undefined or null, the preview Modal header displays "draft"
* Otherwise, displays "published"

## Steps to test
1. Go to the admin dashboard
2. If a book review does not exist, publish one 
3. click the preview button (eye), in the top left corner: "Previewing" should say published if the review is published
4. Since draft reviews cannot be previewed yet, change the data field for published_at to null and tests that the book review says "Draft"

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have run docker-compose up and my project compiled
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/PreviewReview/PreviewReviewModal.tsx [hunk 1]
```diff
@@ -44,7 +44,14 @@ const PreviewReviewModal = ({
     }
     return "";
   };
-
+  // modalState returns "Published" if published_at has a value and "Draft" if it is null
+  const modalState = () => {
+    if (review.publishedAt == null) {
```
Comment (susheelk) [inline frontend/src/components/PreviewReview/PreviewReviewModal.tsx:49]: I think you also need to check for `undefined` here. Both `undefined` and `null` evaluate to `false`, so I think you can just do something like `if(!review.publishedAt) {...`

**Review by susheelk — CHANGES_REQUESTED** [review]
Nice work! :smile: Just added a minor comment.

File: frontend/src/components/PreviewReview/PreviewReviewModal.tsx (hunk 1 above)
Comment (helenellyx) [reply frontend/src/components/PreviewReview/PreviewReviewModal.tsx:49]: Thank you! 
I think the `==` uses the juggling-check method so it checks for `null` and `undefined` at the same time, if I say `==null`. (https://www.delftstack.com/howto/typescript/check-for-both-null-and-undefined-in-typescript/#use-the-and-operators-to-check-for-both-null-and-undefined-in-typescript). 

Let me know if it is wrong though!

Comment (susheelk) [reply frontend/src/components/PreviewReview/PreviewReviewModal.tsx:49]: Looks like you're right! :smile: Taught me something lol

**Review by susheelk — APPROVED** [review]
Code looks good, tested and functionality looks good! Go ahead and merge!
Awesome work :smile:

**Pull request merged**